### PR TITLE
Display graceful login error when the backend down

### PR DESCRIPTION
**Pull request by dionyziz**
This patch displays a graceful login error and re-enabled the login button in
case the backend service is down.

### CODE REVIEW

File: frontend/src/components/welcome/common/LoginForm.js [hunk 1]
```diff
@@ -81,6 +89,8 @@ class LoginForm extends Component {
                             onChange={this.handleInputChange}
                         />
                         {this.state.showCredentialsInvalid ? <Tooltip id="tooltip" placement="bottom" className="in">Λάθος email ή password!</Tooltip> : ''}
+                        {this.state.unknownLoginError ? <Tooltip id="tooltip" placement="bottom"
+                        className="in">Προσπάθησε ξανά αργότερα</Tooltip> : ''}
```
Comment (dimkarakostas) [inline frontend/src/components/welcome/common/LoginForm.js]: Do the same for the signup form

File: backend/backend/settings.py [hunk 2]
```diff
@@ -27,10 +27,10 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-with open(os.path.join(BASE_DIR, '..', 'config', 'services.json')) as json_services_file:
-    json_services = json.load(json_services_file)
+with open(os.path.join(os.path.dirname(BASE_DIR), 'config', 'services.json')) as json_services_file:
+    service_data = json.load(json_services_file)
```
Comment (dimkarakostas) [inline backend/backend/settings.py]: I get "AttributeError: 'Settings' object has no attribute 'service_data'", use uppercase as stated [here](https://docs.djangoproject.com/en/1.11/topics/settings/#creating-your-own-settings).

File: backend/unimeet/views.py [hunk 3]
```diff
@@ -7,9 +7,6 @@
 from django.conf import settings
 from models import User
 
-with open(os.path.join(os.path.dirname(settings.BASE_DIR), 'config', 'services.json'), 'r') as f:
```
Comment (dimkarakostas) [inline backend/unimeet/views.py]: Also remove unused os import

File: backend/unimeet/views.py [hunk 4]
```diff
@@ -19,7 +16,7 @@ def get_schools(request):
 @csrf_exempt
 def signup(request):
     signup_parameters = json.loads(request.body.decode('utf-8'))
-    email = signup_parameters['email']
+    email = signup_parameters.get('email')
```
Comment (dimkarakostas) [inline backend/unimeet/views.py]: Why are these changes necessary?

Comment (dionyziz) [reply backend/unimeet/views.py]: In case the user does not pass any parameters, we don't want it to crash with an exception (status code 500) – instead it should fail gracefully by behaving as if the email was set to `None` (status code 400).

**Comment by dionyziz** [comment]
PTAL

**Comment by dimkarakostas** [comment]
LGTM, thanks
